substrate/in-instructions/pallet/src/lib.rs

```diff
@@ -70,10 +80,110 @@ pub mod pallet {
     StorageMap<_, Blake2_256, NetworkId, BlockHash, OptionQuery>;
 
   impl<T: Config> Pallet<T> {
-    fn execute(instruction: InInstructionWithBalance) -> Result<(), ()> {
+    fn execute(instruction: InInstructionWithBalance) -> DispatchResult {
```

Comment:
This type may overlap with the `DispatchResult` type alias, but it's not used as a `DispatchResult`. `execute_batch` (the dispatchable) gets it, performs is_err, but won't `?`.

Reply:
`?`would return the `execute_batch` but we have multiple in_ins in a batch and `Self::execute()` is called in a for loop per instruction. So returning there would  prevents us from executing remanning instructions in the batch. I thought that is why we deposit instruction failure event if that is what you mean. But yea maybe event can have a `reason` field that holds the `DispatchError`.